contracts/finance/ConfidentialVestingWallet.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import "fhevm/lib/TFHE.sol";
+
+import { IConfidentialERC20 } from "../token/ERC20/IConfidentialERC20.sol";
+
+/**
+ * @title  ConfidentialVestingWallet
+ * @notice This contract offers a simple vesting wallet for ConfidentialERC20 tokens.
+ *         This is based on the VestingWallet.sol contract written by OpenZeppelin.
+ *         see: openzeppelin/openzeppelin-contracts/blob/master/contracts/finance/VestingWallet.sol
+ * @dev    Default implementation is a linear vesting curve.
+ *         To use with the native asset, it is necessary to wrap the native asset to a ConfidentialERC20-like token.
+ */
+abstract contract ConfidentialVestingWallet {
+    /// @notice Emitted when tokens are released to the beneficiary address.
+    event ConfidentialERC20Released();
+
+    /// @notice Beneficiary address.
+    address public immutable BENEFICIARY;
```

Comment:
same here, not sure why you didnt keep OZ's specs with a modifiable owner.

Reply:
ok for this one I understnad because of ACL, yo can keep it immutable then maybe